pkg/controller/machineset/controller.go

```diff
@@ -164,6 +180,49 @@ func (r *ReconcileMachineSet) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, nil
 	}
 
+	var currentFeatureGates featuregates.FeatureGate
+	currentFeatureGates, err := r.featureGateAccessor.CurrentFeatureGates()
+
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("unable to retrieve current feature gates: %w", err)
+	}
+
+	if currentFeatureGates.Enabled(features.FeatureGateMachineAPIMigration) {
+		machineSetCopy := machineSet.DeepCopy()
```

Comment:
Do we need a copy here?

Reply:
yep my bad - not sure what I was thinking here!

Reply:
Although, weirdly removing it is making the tests fail......